raster-foundry/raster-foundry
# Support Organization Creation Requests

## Overview

This set of changes adjusts the status tracking of organizations and allows for any user to request a new organization. This will allow users to request a new organization is created from a platform administrator.

### Checklist

~- [ ] Styleguide updated, if necessary~
- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated - https://github.com/raster-foundry/raster-foundry-api-spec/pull/24
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

![demo-status-filter](https://user-images.githubusercontent.com/898060/41788369-6de93414-7619-11e8-985a-14a9b9c7b056.png)
Shows new status filter and set of organizations with changed status

### Notes

 - This does not adjust the modal or frontend logic
 - This does not send out notifications to platform administrators when an organization has been requested


## Testing Instructions

 * Log in as a platform administrator and create a new organization, this should work as expected
 * Get credentials for both a platform administrator and a platform member
 * Use the following template request to create an organization with each user. Ensure that the platform administrator can set any status they wish on creation while organizations created by members are always set to `REQUESTED`.
```
curl 'http://localhost:9091/api/platforms/31277626-968b-4e40-840b-559d9c67863c/organizations' \
     -H 'Authorization: <JWT-TOKEN>'\
     -H 'Content-Type: application/json;charset=UTF-8'\
     -H 'Accept: application/json, text/plain, */*'\
     --data-binary \
     '{"platformId":"31277626-968b-4e40-840b-559d9c67863c","name":"bcdef","status":"ACTIVE"}'
```
Closes #3583
